### ISSUE TITLE
Row selection unit tests

### ISSUE THREAD

**Issue by martinblostein** [fstpackage/fsttable#30]
Suite of tests consisting of iterated simple row selections and permutations to make sure that the proxy_table updates correctly.

Along these lines:

```
dt <- data.table(x = 1:10)
ft <-fsttable::fst_table(fst::write_fst(dt, tempfile))
ft[2:6][3:2][1] # expect 4
```

("Subissue of #4)

**Comment by MarcusKlik** [comment]
Yes, perfect. And we can add a series of helper functions to compare a `datatableinterface` object with a second `datatableinterface` or an in-memory `data.table` (like [this one](../blob/develop/tests/testthat/test_interface.R#L20-L22)).
